docs/sriov.md

```diff
@@ -1,4 +1,66 @@
-# Enable SR-IOV on the host
+# Architecture
+
+There are multiple components involved to provide SR-IOV support in KubeVirt
+environment.
+
+* [SR-IOV device plugin](https://github.com/intel/sriov-network-device-plugin)
+  discovers available SR-IOV resources, advertises them to Kubernetes resource
+  manager, and allocates them as needed to pods and VMIs. This component
+  doesn't modify host resources, only discovers and allocates them. For VMIs,
+  allocates `vfio` device nodes to pods; also sets environment variables with
+  PCI IDs of allocated devices. These variables are later used by KubeVirt to
+  configure libvirt domain.
+* [SR-IOV CNI plugin](https://github.com/intel/sriov-cni) configures allocated
+  SR-IOV resources before a pod or VMI is started. This component modifies host
+  resources to prepare them to be used inside a pod / VMI. Executed in root
+  network namespace context. Uses netlink and other commands to configure and
+  move SR-IOV VF / PF resources into pod namespaces.
+* [Multus](https://github.com/intel/multus-cni) is a meta-plugin. It will call
+  to SR-IOV CNI plugin when VMI is attached to an SR-IOV interface. It is
+  configured through `NetworkAttachmentDefinition` CRD objects. For SR-IOV, the
+  object annotations should refer to the resource name configured inside device
+  plugin `config.json` configuration file. This reference is used by KubeVirt
+  to automatically fill in `requests` and `limits` sections of `virt-launcher`
+  pods. (The same mechanism is used for other plugins, for example, `bridge`.)
```

Comment:
Also, when using multus + cni + dp we don't require passing the pci address of the device to the cni plugin, because multus will find that from the devices advertised by the device plugin.

Reply:
I'm not sure about adding this note, this seems to refer to a practice that is not used in actual kubevirt environment. But let me know which specific change to the text you had in mind, maybe seeing it in context would help me understand what you mean to add here.

Reply:
Why not? I am referring to what's stated [here](https://github.com/intel/multus-cni/tree/master/examples#passing-down-device-information), and to the fact that having the resourceName annotation is enough to have one device passed. 

It's not super important and we can survive without that information, but in case I'd go with something along the line of:

If used alone, the cni plugin would need the pci address of the device we want to use inside the pod (and then, inside the vm). By using it in combination with the device plugin and multus and using the `resourceName` annotation, the pod will receive a device from the pool allocated by the device plugin under the same resource.